src/tally.F90

```diff
@@ -1173,6 +1230,166 @@ subroutine score_general_ce(p, t, start_index, filter_index, i_nuclide, &
+          end if
+        end if
+
+      case (SCORE_FISS_Q_BETAS)
+
+        if (p % type /= NEUTRON) cycle SCORE_LOOP
+
+        score = ZERO
+
+        if (t % estimator == ESTIMATOR_ANALOG) then
+          if (survival_biasing) then
+            ! No fission events occur if survival biasing is on -- need to
+            ! calculate fraction of absorptions that would have resulted in
+            ! fission scaled by Q-value
+            associate (nuc => nuclides(p % event_nuclide))
+              if (micro_xs(p % event_nuclide) % absorption > ZERO .and. &
+                   allocated(nuc % fission_q_betas)) then
+                score = p % absorb_wgt &
+                     * nuc % fission_q_betas % evaluate(p % last_E) &
+                     * micro_xs(p % event_nuclide) % fission &
+                     / micro_xs(p % event_nuclide) % absorption * flux
+              end if
+            end associate
+          else
+            ! Skip any non-absorption events
+            if (p % event == EVENT_SCATTER) cycle SCORE_LOOP
+            ! All fission events will contribute, so again we can use
+            ! particle's weight entering the collision as the estimate for
+            ! the fission energy production rate
+            associate (nuc => nuclides(p % event_nuclide))
+              if (allocated(nuc % fission_q_betas)) then
+                score = p % last_wgt &
+                     * nuc % fission_q_betas % evaluate(p % last_E) &
+                     * micro_xs(p % event_nuclide) % fission &
+                     / micro_xs(p % event_nuclide) % absorption * flux
+              end if
+            end associate
+          end if
+
+        else
+          if (t % estimator == ESTIMATOR_COLLISION) then
+            E = p % last_E
+          else
+            E = p % E
+          end if
+
+          if (i_nuclide > 0) then
+            if (allocated(nuclides(i_nuclide) % fission_q_betas)) then
+              score = micro_xs(i_nuclide) % fission * atom_density * flux &
+                      * nuclides(i_nuclide) % fission_q_betas % evaluate(E)
+            end if
+          else
+            if (p % material /= MATERIAL_VOID) then
+              do l = 1, materials(p % material) % n_nuclides
+                atom_density_ = materials(p % material) % atom_density(l)
+                i_nuc = materials(p % material) % nuclide(l)
+                if (allocated(nuclides(i_nuc) % fission_q_betas)) then
+                  score = score + micro_xs(i_nuc) % fission * atom_density_ &
+                          * flux &
+                          * nuclides(i_nuc) % fission_q_betas % evaluate(E)
+                end if
+              end do
+            end if
+          end if
+        end if
+
+      case (SCORE_Q_ELASTIC)
+
+        if (p % type /= NEUTRON) cycle SCORE_LOOP
+
+        ! Skip any non-elastic scatter events
+        if (p % event_MT /= ELASTIC) cycle SCORE_LOOP
+
+        score = p % wgt * (p % last_E - p % E)
```

Comment:
Agreed. I'm not attached to what I put in - I just wanted sometime to start tallying the energy deposition from different sources.  I look forward to reading those papers.